### PR TITLE
[ENH] Add pull request template

### PR DESCRIPTION
**Pull request by arnaudbore**

### CODE REVIEW

File: docs/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# Checklist:
+
+- [ ] My code follows the style guidelines of this project (mostly pep8 compliant)
+- [ ] I have performed a self-review of my code
+- [ ] I have commented my code, particularly in hard-to-understand areas
+- [ ] I have made corresponding changes to the documentation
+- [ ] My changes generate no new warnings
+- [ ] I moved all functions from the script file (except the argparser and main) to scilpy modules
+- [ ] I have added tests that prove my fix is effective or that my feature works
+- [ ] New and existing unit tests pass locally with my changes
```
Comment (AlexVCaron) [inline docs/pull_request_template.md:30]: This means we have to document running tests better. Especially how to scope which tests to run, where to check for logs, warnings, coverage, etc.

File: docs/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+## Provide data and command line to test
+
+...
+
+# Checklist:
+
+- [ ] My code follows the style guidelines of this project (mostly pep8 compliant)
```
Comment (AlexVCaron) [inline docs/pull_request_template.md:23]: Maybe add a check that all relevant citations have been added to scripts, modules and functions docstrings and descriptions.

Comment (AlexVCaron) [inline docs/pull_request_template.md:23]:
```suggestion
- [ ] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
```

Comment (AlexVCaron) [reply docs/pull_request_template.md:23]: We could replace the link with scil documentation (linting with autopep8 should be described there if it is not already)

File: docs/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
+# Quick description
+
+Please include a summary of the changes and the related issue or the improvement.
```
Comment (AlexVCaron) [inline docs/pull_request_template.md:3]:
```suggestion
Please include a summary of the changes and the related issue(s) or improvement(s). 
```

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
Few comments, very nice boilerplate ! I think with that we need to also improve dev documentation on scil-documentation. Things like testing, linting and others need to be explained clearly if we want people checking all those boxes for real.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

File: docs/pull_request_template.md (hunk 2 above)
Comment (arnaudbore) [reply docs/pull_request_template.md:23]: Let's use this link for now and will improve later.

File: docs/pull_request_template.md (hunk 1 above)
Comment (arnaudbore) [reply docs/pull_request_template.md:30]: We'll see later, this template is there as a reminder of everything that has to be done when you submit a PR.
Don't need to over think this right now, I think.

**Comment by arnaudbore** [comment]
Build passed ! Good Job :beers: !

File: docs/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] This change requires a documentation update
+
+## Provide data and command line to test
+
+...
+
```
Comment (AntoineTheb) [inline docs/pull_request_template.md:20]: _Maybe_ add a section for screenshots if relevant ?

**Review by AntoineTheb — APPROVED** [review]
LGTM ! Thanks for this @arnaudbore

File: docs/pull_request_template.md (hunk 2 above)
Comment (EmmaRenauld) [reply docs/pull_request_template.md]: I can put this on my to-do list.

**Review by EmmaRenauld — APPROVED** [review]

**Pull request merged**